File: CRUD/Java/app/src/main/java/com/fj/tpamobile/crud_java/adapters/NewsAdapter.java
package com.fj.tpamobile.crud_java.adapters;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.fj.tpamobile.crud_java.R;
import com.fj.tpamobile.crud_java.activities.AddNewsActivity;
import com.fj.tpamobile.crud_java.data.Utils;
import com.fj.tpamobile.crud_java.models.News;

import java.util.ArrayList;

public class NewsAdapter extends RecyclerView.Adapter<NewsAdapter.ViewHolder> {

    private Context context;
    private ArrayList<News> news;

    public NewsAdapter(Context context, ArrayList<News> news) {
        this.context = context;
        this.news = news;
    }

    @NonNull
    @Override
    public NewsAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_news, viewGroup, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull NewsAdapter.ViewHolder viewHolder, final int i) {
        viewHolder.txtTitle.setText(news.get(i).getTitle());
        viewHolder.txtContent.setText(news.get(i).getContent());

        viewHolder.cardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(context, AddNewsActivity.class);
                intent.putExtra(Utils.NewsId, news.get(i).getId());
                context.startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return news.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        TextView txtTitle, txtContent;
        CardView cardView;

        ViewHolder(@NonNull View itemView) {
            super(itemView);

            txtTitle = itemView.findViewById(R.id.txt_title);
            txtContent = itemView.findViewById(R.id.txt_content);
            cardView = itemView.findViewById(R.id.cv_news);
        }
    }
}

File: CRUD/Kotlin/app/src/main/java/com/fj/tpamobile/crud_kotlin/adapters/NewsAdapter.kt
package com.fj.tpamobile.crud_kotlin.adapters

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.fj.tpamobile.crud_kotlin.R
import com.fj.tpamobile.crud_kotlin.models.News
import kotlinx.android.synthetic.main.item_news.view.*

class NewsAdapter(
    private val context: Context,
    private val news: ArrayList<News>,
    private val listener: (News) -> Unit
) : RecyclerView.Adapter<NewsAdapter.ViewHolder>() {

    override fun onCreateViewHolder(p0: ViewGroup, p1: Int): NewsAdapter.ViewHolder =
        ViewHolder(LayoutInflater.from(context).inflate(R.layout.item_news, p0, false))

    override fun getItemCount(): Int = news.size

    override fun onBindViewHolder(p0: NewsAdapter.ViewHolder, p1: Int) =
            p0.bindNews(news[p1], listener)

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        fun bindNews(news: News, listener: (News) -> Unit){
            itemView.txt_title.text = news.title
            itemView.txt_content.text = news.content

            itemView.cv_news.setOnClickListener {
                listener(news)
            }
        }
    }
}